### ISSUE TITLE
PDBRX report: "edit the system segmentation" 

### ISSUE THREAD

**Issue by cjeong73** [madscatt/zazzie#116]
- Input pdb file, 4F87.pdb has 4 chains(A,B,C, and D) with missing residues in each chain.

- **PDBSCAN** reported the detailed information of sequences including missing residues. According to the log of PDBSCAN, fasta sequence should be...

A>
mskinvnvenVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
B>
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
C>
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
D>
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk

- **In case of Non-default mode, PDBRX in dev branch** generated the sequences as below. Note Chain B,C had no missing residues while chain A, D have missing residue(s) at one chain end. **Another issue is that the output structure was made based on this sequence info**. 

Current sequences (lowercase indicates residues not in coordinates): 
A:
mskinvnvenVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
C:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
B:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
D:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk

- **PDBRX in jetstream resulted in different sequences.** 

A:
VSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
C:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
B:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK
D:
NVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMK

**Comment by cjeong73** [comment]
**In terminal_edit_options module of preprocess.py, I replace mol.segname_info.sequence_to_fasta with mol.chain_info.sequence_to_fasta to get fasta sequence.**

            print("Current sequences (lowercase indicates residues not in coordinates): ")

            for segname in seq_segnames:
              #  seq = mol.segname_info.sequence_to_fasta(
                seq = mol.chain_info.sequence_to_fasta(
                    segname, missing_lower=True)
                print(segname + ':')
                print(seq)

**Then, non-default mode printed out the segmentation info correctly as below. Need to find out the occurrence of this error in the codes.**

Current sequences (lowercase indicates residues not in coordinates): 
A:
mskinvnvenVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
C:
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
B:
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk
D:
mskinvnveNVSGVQGFLFHTDGKESYGYRAFINGVEIGIKDIETVQGFQQIIPSINISKSDVEAIRKAMKk

**Comment by madscatt** [comment]
While this gives the correct output it is only for a "print" statement.

**Comment by cjeong73** [comment]
Yes, it is only for print statement. Then, I tested what happens if segname_info is replaced with chain_info in all pdbrx codes of dev branch at onsager. This change corrected the output psf and pdb following the sequence info of pdbscan at least for dev branch codes.

**Comment by madscatt** [comment]
That is a possible solution, perhaps a better solution (as segname/segment is the currency involved) is to update the segname_info object.  I am very uneasy with finding solutions by replacing bits of code with out understanding the consequences.

**Comment by madscatt** [comment]
See issue report in #117

**Comment by dww100** [comment]
@cjeong73 What is the status of this post the fixes for #117?

**Comment by madscatt** [comment]
Current take-home: this should stay in the reproduction/characterization phase before we change code. The issue appears to be a data transfer problem between chain-level sequence truth (chain_info) and segment-level sequence state (segname_info), and PDBRx uses segname_info as the working object for display, edits, gap descriptions, PyRosetta building, and output. So replacing segname_info with chain_info in PDBRx may make one symptom look better, but it risks hiding the actual mapping bug.\n\nSuggested first step only:\n- Reproduce with the minimal s.pdb fixture from issue #117 and, separately, the fuller 4F87 case from this issue.\n- For each fixture, print/compare chain_info.sequence, segname_info.sequence, sequence_to_fasta(..., missing_lower=True), and seqs_for_completion() immediately after mol.run_scan().\n- Establish expected invariants before attempting a fix: terminal missing residues from chain_info should be preserved correctly in segname_info for each mapped segment, and PDBRx gap descriptions should reflect that corrected segment sequence state.\n\nImportant caution: we do not have a deep PDBScan/PDBRx test suite right now. Before safely addressing this issue, the test suite needs to be extended with small targeted regression fixtures/probes for this exact chain-to-segname sequence transfer behavior. Without that, this is the kind of bug where a local-looking fix can produce plausible but wrong structures downstream.\n\nRecommended next action: add focused reproduction tests/probes first, then revisit the scanner.py get_segment_sequence_info mapping logic once the expected behavior is locked down.

**Comment by madscatt** [comment]
Confirmed valid. The non-default segmentation prompt is expected PDBRx behavior, but the reported sequence mismatch is a real chain_info -> segname_info transfer problem. Current scan-only probes still show 4F87 preserving terminal missing residues in chain_info while segname_info loses them for interior chains. This can affect displayed sequences, user sequence edits, PyRosetta gap completion, and final PDB/PSF output.

I would not replace PDBRx segname_info usage with chain_info globally. That may fix simple chain==segment cases, but it risks breaking edited segmentation and BIOMT/segment workflows, where segment/segname is the working PDBRx object.

Recommended path: keep this in characterization/fix scope for PDBScan, add focused s.pdb and 4F87 regression coverage for chain-to-segname sequence transfer, then fix get_segment_sequence_info so segname_info is correct before PDBRx consumes it.
